### PR TITLE
Update to be consistent with other Rails 6 upgrades 

### PR DESCRIPTION
**Pull request by benbaumann95**
This PR makes this app consistent with other rails 6 upgrades we are doing by fully upgrading the application.

This includes pinning the Rails version at the top, switching to use config.load_defaults 6.0 and updating the remaining minor dependencies.

### CODE REVIEW

**Comment by benbaumann95** [comment]
> Mostly looking good.
> 
> * [914e2c6](https://github.com/alphagov/support/commit/914e2c60d3052cde4532bd35546955d15ba4343d)
>   I'm not sure if it's a requirement or just a good to have, but I've been putting a reference to that Andrew commit: [alphagov/datagovuk_find@0150923](https://github.com/alphagov/datagovuk_find/commit/0150923813ec45293c6108640c9846f37ae2afa6)
> * [a415942](https://github.com/alphagov/support/commit/a415942241bd518ab5cd46d075c8b29fb9a9762a) and [d3e21f3](https://github.com/alphagov/support/commit/d3e21f3ffd32f874c9760ee022f4722499d5e0ae)
>   I'd squash those two commits together and just say "Update Rails config to 6.0.3.2"
> * [e146e21](https://github.com/alphagov/support/commit/e146e210bc326a95e878e7a234c5833f763b647d)
>   could you please include the reason on the first line? Something like "Add GDS acronym inflection for Zeitwerk autoloader"
>   Also, it seems like the previous commit would leave the app broken since this is being fixed here, and we should try to avoid that. Could you please either squash this one on the previous or amend the previous to use classic autoloader?
> * [c4bdf90](https://github.com/alphagov/support/commit/c4bdf90bc9b33e1c460dcddbadbabb2845545450)
>   Could you please refer to the problem instead of the warning?

For 914e2c6 I guess it was already in the format we agreed on so I didn't think to add a comment. But happy to add the reasoning to give a wider context although might be confusing cos I didn't actually do anything to it?

Happy to adjust to your other commit suggestions

File: config/application.rb [hunk 1]
```diff
@@ -17,10 +17,23 @@
...
+    config.action_controller.per_form_csrf_tokens = false
+
+    # Enable origin-checking CSRF mitigation.
+    config.action_controller.forgery_protection_origin_check = false
+
+    # Make Ruby 2.4 preserve the timezone of the receiver when calling `to_time`.
+    ActiveSupport.to_time_preserves_timezone = false
+
+    # Make `form_with` generate non-remote forms.
+    config.action_view.form_with_generates_remote_forms = false
```
[the remark this answers is not in the document]
Comment (benbaumann95) [reply config/application.rb:37]: Just copied these comments over from a `new_framework_defaults.rb` file.
Might make sense to update but need to check whether those comments are default by Rails or not.
In the case it is, i think it usually has truthy statements above the configuration whether you switch it from true to false.

File: config/application.rb [hunk 2]
```diff
@@ -1,8 +1,12 @@
 require_relative "boot"
 
-# require 'rails/all'
+require "rails"
+
+require "active_model/railtie"
+require "active_job/railtie"
 require "action_controller/railtie"
 require "action_mailer/railtie"
+require "action_view/railtie"
```
[the remark this answers is not in the document]
Comment (benbaumann95) [reply config/application.rb:9]: Maybe not, worth checking 👍

**Comment by EGiataganas** [comment]
Hey @benbaumann95 this looks great, I've compared the PR changes with [railsdiff][1] and there are some changes that could also be done:

- Do we need to explicitly add the local dev host in the `config/environments/development.rb`?
- In the `config/environments/test.rb` the `config.cache_classes = true` has changed to `false`
- Update `app/views/layouts/application.html.erb`to include `csp_meta_tag` after `csrf_meta_tags`. Maybe it will be good to have a `csrf_meta_tags`, `csp_meta_tag`, `stylesheet_link_tag` and then `javascript_include_tag`

[1]: http://railsdiff.org/5.2.4.3/6.0.3.2

**Review by cdccollins — APPROVED** [review]

**Pull request merged**